packages/core/src/browser/storage-service.ts

```diff
@@ -57,7 +61,17 @@ export class LocalStorageService implements StorageService {
 
     setData<T>(key: string, data?: T): Promise<void> {
         if (data !== undefined) {
-            this.storage[this.prefix(key)] = JSON.stringify(data);
+            try {
+                if (!this.fullstorage) {
```

Comment:
ATM there is no way for ```fullstorage``` to go back to a ```false``` value, until the client is restarted. If e.g. the user cleans-up the local storage without restarting the client, we would still skip attempting to save in the local storage.

Instead we might try to save to local storage every time, but only report that we failed once.